pytest_hot_reloading/plugin.py

```diff
@@ -85,6 +87,48 @@ def _jurigged_logger(x: str) -> None:
     """
 
 
+OrigFunctionDefinition = jurigged_codetools.FunctionDefinition
+
+
+class NewFunctionDefinition(OrigFunctionDefinition):
+    def reevaluate(self, new_node, glb):
+        new_node = self.apply_assertion_rewrite(new_node, glb)
+        return super().reevaluate(new_node, glb)
+
+    def apply_assertion_rewrite(self, ast_func, glb):
```

Comment:
This is a copy and paste from the tail end of the function `_pytest.assertion.rewrite.AssertionRewriter.run`. Their logic is not properly broken up to where a smaller slice can be borrowed (operates at a module level).